### DISCUSSION TITLE
Example of using custom fetch

### DISCUSSION THREAD

**Discussion by jludwiggreenaction** [openapi-ts/openapi-typescript#1516] [Q&A] [answered]
I'd like to use `axios` as my fetch client, but I'm having trouble understanding what interface I should be implementing since it's typed as `any`.

Is there any documentation or example I could be directed towards to get started with this? Thanks.

**Comment by drwpow** [comment, marked as answer]
Oh the `any` type could probably be improved; that’s just to prevent errors with 3rd-party libraries.

It implements the `fetch()` API perfectly, so if you wanted to connect it to Axios, you’d have to add a fetch-compatible wrapper, like:

```ts
const client = createClient({
  fetch(url: string, options: RequestInit) {
    return axios({ url, ...options });
  }
});
```

[Here is the line of code](https://github.com/drwpow/openapi-typescript/blob/main/packages/openapi-fetch/src/index.js#L69) that gets called at runtime.
